web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx

```diff
@@ -0,0 +1,240 @@
+  onDelete: () => Promise<void>;
+  onCancel: () => void;
+}) {
+  const theme = useTheme();
+  const [convertAttempt, convertToRole] = useAsync(
+    async (yaml: string): Promise<RoleWithYaml | null> => {
+      if (resources.status === 'creating' || !resources.item) {
+        return null;
+      }
+      return {
+        yaml,
+        object: await yamlService.parse<Role>(YamlSupportedResourceKind.Role, {
+          yaml,
+        }),
+      };
+    }
+  );
+
+  const originalContent = resources.item?.content ?? '';
+  useEffect(() => {
+    convertToRole(originalContent);
+  }, [originalContent]);
+
+  return (
+    <Flex flex="1">
+      <Flex
+        flexDirection="column"
+        borderLeft={1}
+        borderColor={theme.colors.interactive.tonal.neutral[0]}
+        backgroundColor={theme.colors.levels.surface}
+        width="700px"
+      >
+        {convertAttempt.status === 'processing' && (
+          <Flex
+            flexDirection="column"
+            alignItems="center"
+            justifyContent="center"
+            flex="1"
+          >
+            <Indicator />
+          </Flex>
+        )}
+        {convertAttempt.status === 'error' && (
+          <Danger>{convertAttempt.statusText}</Danger>
+        )}
+        {convertAttempt.status === 'success' && (
+          <RoleEditor
+            originalRole={convertAttempt.data}
+            onCancel={onCancel}
+            onSave={onSave}
+            onDelete={onDelete}
+          />
+        )}
+      </Flex>
+      <Flex flex="1" alignItems="center" justifyContent="center" m={3}>
+        <Box>
+          <H1 mb={2}>Teleport Policy</H1>
+          <Flex mb={4}>
+            <Box flex="1" width="min-content">
+              <P>
+                Teleport Policy will visualize resource access paths as you
+                create and edit roles so you can always see what you are
+                granting before you push a role into production.
+              </P>
+            </Box>
+            <Flex flex="0 0 auto" ml={6} alignItems="start">
+              <ButtonLockedFeature noIcon py={0} width={undefined}>
+                Contact Sales
+              </ButtonLockedFeature>
+              <ButtonSecondary
+                as="a"
+                href="https://goteleport.com/platform/policy/"
+                target="_blank"
+                ml={2}
+              >
+                Learn More
+              </ButtonSecondary>
+            </Flex>
+          </Flex>
+          <Flex
+            flexDirection="column"
+            bg={theme.colors.levels.surface}
+            borderRadius={3}
+            // I found hardcoding the width of the container to be the only way
+            // to play well with the StepSlider below. Trying to use implicit
+            // width in combination with `width: min-content` inside a
+            // StepSlider child caused a miscalculation of child's height.
+            width={promoImageWidth + 2 * 2}
```

Comment:
Regarding responsiveness, here's something you could try:

<details>
<summary>Patch</summary>
Copy then `pbpaste | git apply`

```diff
diff --git a/web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx b/web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx
index 4b32c3e04b..ff05755e2d 100644
--- a/web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx
+++ b/web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx
@@ -109,17 +109,17 @@ export function RoleEditorAdapter({
         )}
       </Flex>
       <Flex flex="1" alignItems="center" justifyContent="center" m={3}>
-        <Box>
+        <Box maxWidth={782} minWidth={300}>
           <H1 mb={2}>Teleport Policy</H1>
-          <Flex mb={4}>
-            <Box flex="1" width="min-content">
+          <Flex mb={4} gap={4} flexWrap="wrap" justifyContent="space-between">
+            <Box flex="1" minWidth="30ch">
               <P>
                 Teleport Policy will visualize resource access paths as you
                 create and edit roles so you can always see what you are
                 granting before you push a role into production.
               </P>
             </Box>
-            <Flex flex="0 0 auto" ml={6} alignItems="start">
+            <Flex flex="0 0 auto" alignItems="start">
               <ButtonLockedFeature noIcon py={0} width={undefined}>
                 Contact Sales
               </ButtonLockedFeature>
@@ -137,24 +137,13 @@ export function RoleEditorAdapter({
             flexDirection="column"
             bg={theme.colors.levels.surface}
             borderRadius={3}
-            // I found hardcoding the width of the container to be the only way
-            // to play well with the StepSlider below. Trying to use implicit
-            // width in combination with `width: min-content` inside a
-            // StepSlider child caused a miscalculation of child's height.
-            width={promoImageWidth + 2 * 2}
           >
             <Box
               border={2}
               borderRadius={3}
               borderColor={theme.colors.interactive.tonal.neutral[0]}
             >
-              {/* Note: image dimensions hardcoded to prevent UI glitches while
-                  loading. */}
-              <Image
-                src={tagpromo}
-                width={promoImageWidth}
-                height={promoImageHeight}
-              />
+              <Image src={tagpromo} width="100%" />
             </Box>
             <StepSlider flows={promoFlows} currFlow="default" />
           </Flex>
@@ -164,9 +153,6 @@ export function RoleEditorAdapter({
   );
 }
 
-const promoImageWidth = 782;
-const promoImageHeight = 401;
-
 const promoFlows = {
   default: [PromoPanel1, PromoPanel2],
 };

```
</details>

The key is setting some max width for the entire box, so that the text doesn't just take all possible width.

I'm not 100% sure if this satisfies all your requirements, but it looks good enough IMHO.

https://github.com/user-attachments/assets/00a7f9fa-8dd1-4b0a-a5a8-1f7bf0f0a923

Reply:
Of course! It seems so obvious now. Thanks!

Reply:
I was actually surprised that we don't have default responsive styles for `<img>`. But I suppose we don't use images that often.

Reply:
I think it was supposed to be `width: 100%; height: auto`, but in this case it doesn't matter because of that `max-width` on the parent.

https://stackoverflow.com/questions/15458650/make-an-image-responsive-the-simplest-way